ocf_datapipes/load/gsp.py

```diff
@@ -0,0 +1,103 @@
+import datetime
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("open_gsp")
+class OpenGSPIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        gsp_pv_power_zarr_path: str,
+        gsp_id_to_region_id_filename: str,
+        sheffield_solar_region_path: str,
+        threshold_mw: int = 0,
+        sample_period_duration: datetime.timedelta = datetime.timedelta(minutes=30),
+    ):
```

Comment:
Good liniting. 

i prefer just using `timedelta` rather than `datetime.timedetla`, i think if we import datetime, then it imports the whole thing. plus keeps the code a bit neater. 

Need to add description what these things are